d3rlpy/base.py

```diff
@@ -336,15 +340,27 @@ def fit(self,
             if new_transitions:
                 transitions = env_transitions + new_transitions
 
-            indices = np.random.permutation(np.arange(len(transitions)))
```

Comment:
I'd like you to rewrite shuffling like this
```
if shuffle:
    indices = np.random.permutation(np.arange(len(transitions)))
else:
    indices = np.random.arange(len(transitions)
```
Then, the rest of the iterating thins remains the same.

Reply:
Hi thanks, I think I do not understand, do you mean I should rollback all the batch creation process to keep the building of the list of transitions explicitly without using a generator?

That is to preserve this code?:
```
            n_iters = len(transitions) // self.batch_size
            range_gen = trange(n_iters) if show_progress else range(n_iters)
            for itr in range_gen:
                # pick transitions
                sampled_transitions = []
                head_index = itr * self.batch_size
                for index in indices[head_index:head_index + self.batch_size]:
                    sampled_transitions.append(transitions[index])
```